app/domain/breadcrumb_trail.rb

```diff
@@ -1,29 +1,38 @@
 class BreadcrumbTrail
-  def self.build(item, category_tree)
-    categories = case item
-      when Core::Category
-        if item.parent_id.present?
-          RootToNodePath.build(item, category_tree)[0...-1]
-        else
-          [HomeCategory.new]
-        end
-      when ToolCategory
-        RootToNodePath.build(item, category_tree)
-      when Core::StaticPage
+  def self.build(item, category_tree = nil)
+    case item
+    when Mas::Cms::Category
+      # recurse and find categories instead of passing in category_tree?
+      if item.parent_id.present?
+        build_trail(item.parent_id)
+      else
+        [HomeCategory.new]
+      end
+    when ToolCategory
+      RootToNodePath.build(item, category_tree)
+    when Core::StaticPage
+      [HomeCategory.new]
+    when Core::NewsArticle
+      [HomeCategory.new, NewsCategory.new]
+    else
+      if item.categories.empty?
```

Comment:
as above, perhaps move this if block into it's own smaller method

Reply:
Sorting out `BreadcrumbTrail` anymore than I have done is probably a bit of a rabbit hole in-fact I almost reverted this change out. I think probably the best time to attempt to sort out this class would be when/if we remove the category methods in application controller.

In terms of moving this to a method right now I wanted to keep it somewhat similar to the original version till we can take on the whole class. Likely when we do so this where clause will reduce significantly (or even go completely) and be less of an issue.

Reply:
My only concern to that would be the 

>  when/if we remove the category methods in application controller

*if* we don't end up doing that work in the application controller, then it's more than likely this work won't get done and this code is left, in my eyes, as tech debt. 

When I currently look at the switch statements, I find it quite hard to follow and think it would be slightly easier if the `if` blocks were moved down the class into their own small methods.

I haven't looked at the current backlog of tickets around this, so my question would be, Do you see this work being done as part of another ticket in the very near future?